File: basic/3986/3986.py
import sys

n = int(sys.stdin.readline())

cnt = 0
for i in range(n):
    s = sys.stdin.readline().strip()
    stack = []
    for c in s:
        if len(stack) and stack[-1] == c:
            stack.pop()
        else:
            stack.append(c)
    if len(stack) == 0:
        cnt += 1
sys.stdout.write(str(cnt)+"\n")

File: basic/3986/3986.cpp
#include <bits/stdc++.h>
using namespace std;

int n, cnt;
string s;

int main(){
    cin >> n;
    for(int i=0; i<n; i++){
        cin >> s;
        stack<char> str;
        for (char c : s){
            if (str.size() && str.top() == c) str.pop();
            else str.push(c);
        }
        if (str.size() == 0) cnt++;
    }
    cout << cnt << "\n";
    return 0;
}